R/variable-as-methods.R

```diff
@@ -0,0 +1,133 @@
+#' function (`as.Text`, `as.Numeric`, and `as.Categorical` respectively) that 
+#' takes the input given as `x`, and makes a new derived variable that is now of 
+#' the type specified. See below for detailed examples.
+#' 
+#' For `as.Text` and `as.Numeric`, aliases to the R-native functions 
+#' `as.character` and `as.numeric` are provided for convenience.
+#'
+#' @param x a Crunch variable to derive and convert to a new type
+#' @param format for `as.Datetime`, when the variable in `x` is a text or 
+#' categorical variable, `format` is the typographical format that the datetime
+#' is already formatted in that needs to be parse from (default: 
+#' \code{"\%Y-\%m-\%d \%H:\%M:\%S"}); for `as.Text` and  `as.Categorical`, is 
+#' the typographical format that the datetime is to be formatted as (e.g. 
+#' "2018-01-08 12:39:57" default: \code{"\%Y-\%m-\%d \%H:\%M:\%S"}).
+#' @param resolution for `as.Datetime`, when the variable in `x` is a numeric 
+#' variable, the resolution of the number (e.g. `"ms"` for milliseconds, `"s"` 
+#' for seconds, etc. see [expressions] for more information about valid values.)
+#' @param offset for `as.Datetime`, when the variable in `x` is a numeric the, a 
+#' character of the offset to count from in the shape "2018-01-08 12:39:57". If 
+#' not supplied, Crunch's default of 1970-01-01 00:00:00 will be used.
+#' @param ... additional arguments for `as.character` and `as.numeric`, ignored when used with 
+#' Crunch variables 
+#'
+#' @return a `VariableDefinition` to be used as the derivation 
+#'
+#' @examples
+#' \dontrun{
+#' # ds$v1 is of type Text
+#' is.Text(ds$v1)
+#' # [1] TRUE
+#' 
+#' # that has strings of numbers
+#' as.vector(ds$v1)
+#' # [1] "32"   "8"    "4096" "1024"
+#' 
+#' # convert this to a numeric variable with the alias `v1_numeric`
+#' ds$v1_numeric <- as.Numeric(ds$v1)
+#' 
+#' # the values are the same, but are now numerics and the type is Numeric
+#' as.vector(ds$v1_numeric)
+#' # [1]   32    8 4096 1024
+#' is.Numeric(ds$v1_numeric)
+#' # [1] TRUE
+#' 
+#' # this new variable is derived, so if new data is appended or streamed, the
+#' # new rows of data will be updated.
+#' is.derived(ds$v1_numeric)
+#' # [1] TRUE
+#' }
+#' @name variable-as-methods
+#' @aliases as.Categorical as.Datetime as.Numeric as.Text
+NULL
+
+
+#' @rdname variable-as-methods
+#' @export
+setMethod("as.Numeric", "CrunchVariable", function (x) {
+    haltIfArray(x, callingFunc = "as.Numeric()")
+    
+    # datetimes are special
+    if (is.Datetime(x)) {
+        return(zfuncExpr("datetime_to_numeric", x))
+    }
+    
+    return(zfuncExpr("cast", x, "numeric"))
+})
+
+#' @rdname variable-as-methods
+#' @export
+setMethod("as.Text", "CrunchVariable",
+          function (x, format = "%Y-%m-%d %H:%M:%S") {
+    haltIfArray(x, callingFunc = "as.Text()")
+    
+    # datetimes are special
+    if (is.Datetime(x)) {
+        return(zfuncExpr("format_datetime", x, list(value = format)))
```

Comment:
I think the default `format` should depend on the `resolution` of the datetime variable.